### PR TITLE
Channel: Add delete function

### PR DESCRIPTION
**Pull request by jak3z**
## About

Just easy way to delete channels

## Status

- [x] These changes have been tested against Discord API or do not contain API change.
- [ ] This PR includes only documentation changes, no code change.
- [ ] This PR introduces some Breaking changes.

### CODE REVIEW

**Comment by invalidCards** [comment]
Why not just use [guild.channels.delete(id)](https://github.com/harmonyland/harmony/blob/main/src/managers/guildChannels.ts#L58-L60)?

**Comment by invalidCards** [comment]
(By the way, if anything, this method should be on the GuildChannel class, as it's not possible to delete non-guild channels, such as (group) DMs.)

**Comment by jak3z** [comment]
> (By the way, if anything, this method should be on the GuildChannel class, as it's not possible to delete non-guild channels, such as (group) DMs.)

Technically you have how to close the channel, but i do not know the difference of this for a bot

> Why not just use [guild.channels.delete(id)](https://github.com/harmonyland/harmony/blob/main/src/managers/guildChannels.ts#L58-L60)?

I did not know I had but, if receive only the channel is kinda weird do `GuildChannel.guild.channels.delete(GuildChannel.id)` but it makes sense

**Comment by invalidCards** [comment]
> Technically you have how to close the channel, but i do not know the difference of this for a bot
You're right, it's the same endpoint.

**Comment by DjDeveloperr** [comment]
Seems reasonable to me. But should be a part of GuildChannel class, not Channel (which includes DM channels that cannot be deleted).

File: src/structures/channel.ts [hunk 1]
```diff
@@ -151,6 +151,11 @@ export class GuildChannel extends Channel {
     this.parentID = data.parent_id ?? this.parentID
   }
 
+  delete(): this {
+    this.client.rest.endpoints.deleteChannel(this.id)
```
Comment (DjDeveloperr) [inline src/structures/channel.ts:155]: Delete needs to be async and you need to await deleteChannel.

**Review by DjDeveloperr — CHANGES_REQUESTED** [review]

File: src/structures/channel.ts (hunk 1 above)
Comment (jak3z) [reply src/structures/channel.ts:155]: oh yeah, forget

**Review by DjDeveloperr — APPROVED** [review]
LGTM, thanks

**Pull request merged**